File: src/canvas/resizeCanvas/resizeCanvas.ts
import never from 'never'
import Input from './Input'
import aspectFit from 'aspect-fit'

const resizeCanvas = ({ canvasRef, containerRef, videoRef }: Input): void => {
  const video = videoRef.current ?? never()
  const canvas = canvasRef.current ?? never()
  const container = containerRef.current ?? never()

  const fit = aspectFit(video.videoWidth, video.videoHeight, container.offsetWidth, container.offsetHeight)
      ;(canvas.width) = fit.width
  ;(canvas.height) = fit.height
}

export default resizeCanvas

File: src/hand_raised/can_camera/camera_success/detector/canvas/resize_canvas.rs
use aspect_fit::{aspect_fit::aspect_fit, scale_size::scale_size, size::Size};

use super::resize_canvas_input::ResizeCanvasInput;

pub fn resize_canvas(input: &ResizeCanvasInput) {
    let video = input.video_ref.current().unwrap();
    let canvas = input.canvas_ref.current().unwrap();
    let container = input.container_ref.current().unwrap();

    let video_scale = Size {
        width: video.video_width(),
        height: video.video_height(),
    };
    let fit = scale_size(
        &aspect_fit::<f64, u32>(
            &video_scale,
            &Size {
                width: container.offset_width() as u32,
                height: container.offset_height() as u32,
            },
        ),
        &video_scale,
    );
    canvas.set_width(fit.width);
    canvas.set_height(fit.height);
}
